src/ScriptBuilder/Saga/SagaDefinitionReader.cs

```diff
@@ -1,13 +1,20 @@
-﻿using System.Linq;
+﻿using System;
+using System.Linq;
 using Mono.Cecil;
+using Mono.Cecil.Cil;
 using NServiceBus.Persistence.Sql;
 using NServiceBus.Persistence.Sql.ScriptBuilder;
 
 static class SagaDefinitionReader
 {
-    public static bool TryGetSqlSagaDefinition(TypeDefinition type, out SagaDefinition definition)
+    public static bool TryGetSagaDefinition(TypeDefinition type, out SagaDefinition definition)
     {
-        ValidateIsNotDirectSaga(type);
+        return TryGetSqlSagaDefinition(type, out definition)
+               || TryInferSagaDefinition(type, out definition);
+    }
```

Comment:
> return TryGetSqlSagaDefinition(type, out definition) || TryInferSagaDefinition(type, out definition);

I think in a few months it'll make us pause for some time, can we come up with better names? maybe something like "TryGetSqlSagaDefinition ... || TryGetCoreSagaDefinition..."